includes/Core/Modules/Modules.php

```diff
@@ -1119,4 +1127,38 @@ function( Module $module ) {
 		);
 	}
 
+	/**
+	 * Gets the recoverable modules.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List of recoverable modules.
+	 */
+	public function get_recoverable_modules() {
+		$recoverable_modules = array();
+		$shareable_modules   = $this->get_shareable_modules();
+
+		foreach ( $shareable_modules as $module ) {
+			$owner_id = $module instanceof Module_With_Owner && $module->get_owner_id();
```

Comment:
@hussain-t this expression results in a boolean rather the value of the owner ID as you would get in JavaScript. This results in the permission checks failing because there is no user with an ID of `true`.

To fix this, we can actually remove the `Module_With_Owner` condition here as all shareable modules are modules with an owner. Instead, we can update the return value of `get_shareable_modules` to be `Module_With_Owner[]` so that IDEs are aware that this method is available on `$module`.

cc: @eugene-manuilov